### ISSUE TITLE
Allow using of different map providers

### ISSUE THREAD

**Issue by Wohlstand** [telegramdesktop/tdesktop#6789]
The issue would be similar to #2787, however, it's for general ability to use any other map provider but Google Maps.

There are examples of why it needs to support other map providers:
* Taste of users: Personally I prefer Yandex Maps for visually better drawing and colors choice
* Details of regions: For example, Yandex Maps has much more details of Russian area (Google Maps is often missing up villages and far places while Yandex showing them up in all details)
* Availability in certain regions: The Google is banned in China, therefore, it's required to use any other map provider but Google Maps to let Chinese people see a map without proxies and VPNs.

You can make some sort of config that will allow a user to manually construct the URL template of the map provider with giving of position coordinates in a certain formats, etc. (for desktop) if you don't want care about every individual map provider yourself. And using any available mobile map application by user's choice (you may notice user if some features wouldn't work in other map provider app).

**Comment by MasterGroosha** [comment]
Just a comment (might be offtopic): on Windows 10, clicking on Geo message opens UWP application "Maps". However, on Windows 10 LTSC, which has no Metro apps, clicking on such message opens Windows's native popup, asking to open Microsoft Store (which is also absent from LTSC).

**Comment by Wohlstand** [comment]
@MasterGroosha, yeah, I can see the same on Windows 8.1 after manual removing of "Maps" UWP application: It will pop-up the same message as you see. I did that at my mother's computer where she didn't like ugly Bing maps thing, and I helped her to get rid of "Maps". It's only after some time after the restart, it will finally open Google Maps in the browser by URL.

**Comment by MasterGroosha** [comment]
@Wohlstand Unfortunately, it doesn't open browser for me :(

So, several suggestions:
1) Add "Copy location" option to Geomessage context menu.
2) Allow to choose maps provider (Yandex, Google, Bing, OpenStreetMaps).

Option "1" is easier of course.

**Comment by camelCaseNick** [comment]
The links from location messages go to Google Maps, but the tiles actually use OpenStreetMap. You can see that, if you know differences in OSM and GMaps or others or even (accidental) trap streets. That is why Telegram needs to attribute OpenStreetMap (see  #6830 for that).
Am I correct in assuming that this issue is about the links and not the in-app tiles?

**Comment by MasterGroosha** [comment]
I think this is still an issue

**Comment by colinner** [comment]
I agree this should be done. I don't see how this could be so hard- it's simply formatting a url with a different domain prefix.

**Comment by VChernenkoRT** [comment]
IMO, Telegram should stop enforcing their users use bloatware
![photo_2021-02-26_00-31-26](https://user-images.githubusercontent.com/66999804/109223895-49325080-77cc-11eb-9260-a1257d697e15.jpg)

**Comment by MasterGroosha** [comment]
@VChernenkoRT your screenshot is from Android version of Telegram. Please, report such issues at https://bugs.telegram.org

**Comment by VasVadum** [comment]
I also believe that for the desktop version of the platform, that it should stop trying to open it using a Microsoft maps feature which tries to open in bing maps but I don't even have that metro app installed so it just opens in a window saying I have no map apps. I wish Telegram would just convert it to a clickable URL upon clicking, instead of requiring me to right click it, copy link, open browser, paste link, hit enter.

Personally I have no problem with Google Maps, I just have a problem with Telegram trying to force me to use Microsoft's map tool because there is no alternative. Windows 10 won't even accept Google Earth Pro as an alternative to bing maps. This is mostly only a problem for desktop on Windows 10 though, I'm sure.

**Comment by proofy** [comment]
What would be happend, if telegram will use the geo URI schema for the link? see https://geouri.org/about/
That would be easy to implement.

**Comment by ilya-fedin** [comment]
Not sure whether it solves the issue but tdesktop should now use the map provider selected in KDE settings on Linux.
